File: src/main/java/common/dto/UserMainMenuDto.java
package common.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserMainMenuDto {

    private Long id;
    private Long pageId;

    private int pageOrder;

    private String pageName;

    private String uiComponent;

    private String routerLink;

    private String icon;

    private List<UserSubMenuDto> subMenus;
}

File: src/app/xgarage/dashboard/dto/usermainmenudto.ts
import { UserSubMenuDto } from './usersubmenudto';

export interface UserMainMenuDto {
  id?: number;
  pageId?: number;
  pageName?: string;
  uiComponent?: string;
  routerLink?: string;
  icon?: string;
  subMenus?: UserSubMenuDto[];
}
